mysql/mysql.go

```diff
@@ -0,0 +1,201 @@
+package mysql
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	// import mysql driver anonymously (just run the init)
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+	nosql "github.com/smallstep/nosql/database"
+)
+
+// DB is a wrapper over *sql.DB,
+type DB struct {
+	db *sql.DB
+}
+
+// Open creates a Driver and connects to the database with the given address
+// and access details.
+func (db *DB) Open(dataSourceName string, opt nosql.Options) error {
+	var err error
+	_db, err := sql.Open("mysql", dataSourceName)
+	if err != nil {
+		return errors.Wrap(err, "error connecting to mysql")
+	}
+	_, err = _db.Exec("CREATE DATABASE IF NOT EXISTS test")
+	if err != nil {
+		return errors.Wrapf(err, "error creating database %s (if not exists)", opt.Database)
+	}
+	db.db, err = sql.Open("mysql", dataSourceName+opt.Database)
+	if err != nil {
+		return errors.Wrapf(err, "error connecting to mysql database")
+	}
+
+	return nil
+}
+
+// Close shutsdown the database driver.
+func (db *DB) Close() error {
+	return errors.WithStack(db.db.Close())
+}
+
+func getQry(bucket []byte) string {
+	return fmt.Sprintf("SELECT nvalue FROM %s WHERE nkey = ?", bucket)
```

Comment:
Is it possible to use `?` instead of `%s`?

Reply:
Same applies to similar methods.

Reply:
No, unfortunately it's not. After the last round of review I actually switched everything to prepared statements that used ?s for all the parameter replacement. I learned the hard way that those only work for "values" (not table or database names).

Reply:
👍